android/src/main/java/com/younics/reachability/ReachabilityAsyncTask.java

```diff
@@ -0,0 +1,84 @@
+package com.younics.reachability;
+
+import android.os.AsyncTask;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+class ReachabilityAsyncTask extends AsyncTask<Void, Void, Boolean> {
+    private Boolean isRunning;
+    private Boolean isReachable;
+    private int timeout = 5000;
+    String hostname = "8.8.8.8";
+    int port = 443;
+    private final ReactApplicationContext reactContext;
+
+    public ReachabilityAsyncTask(String hostname, int port, int timeout, ReactApplicationContext reactContext) {
+        this.reactContext = reactContext;
+
+        if (timeout > 0) this.timeout = timeout;
+        if (hostname != null && !hostname.isEmpty()) this.hostname = hostname;
+        if (port > 0) this.port = port;
+    }
+
+    @Override
+    protected void onPreExecute() {
+        super.onPreExecute();
+
+        this.isRunning = true;
+    }
+
+    @Override
+    public Boolean doInBackground(Void... records) {
+        while (isRunning) {
+            try {
+                try (Socket soc = new Socket()) {
+                    soc.connect(new InetSocketAddress(InetAddress.getByName(hostname), port), timeout);
```

Comment:
Same as before "socket.close();"

Reply:
@Wreit implemented in https://github.com/Younics/react-native-reachability/pull/4/commits/ba9ac1a0abceb66149e3a1059d3ae0a11ac49963